### PR TITLE
Add code formatters in pre-commit hook

### PR DESCRIPTION
**Pull request by justin-sleep**
Not sure if it's something we'd all like, so leaving this as a draft. This is a proof-of-concept PR adding [Ruff](https://github.com/astral-sh/ruff) and [DjHTML](https://github.com/rtts/djhtml) to pre-commit hooks, to enforce a consistent style across the codebase.

### CODE REVIEW

**Comment by rgasper** [comment]
Nice I'm into this

**Comment by BenjaminMalley** [comment]
Nice work! @rgasper what is the merge process? post PR to slack for review?

**Comment by rgasper** [comment]
> 

Sure why not

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  - repo: https://github.com/psf/black
+    rev: 23.7.0
+    hooks:
+      - id: black
+  - repo: https://github.com/rtts/djhtml
+    rev: "3.0.6"
+    hooks:
+      - id: djhtml
+      - id: djcss
+      - id: djjs
```
Comment (rgasper) [inline .pre-commit-config.yaml:27]: could we also add flake8, isort, and some other QoL checks:
```suggestion
      - id: djjs
-   repo: https://github.com/timothycrosley/isort
    rev: 5.12.0
    hooks:
    -   id: isort
        language_version: python3
        args: [
                --multi-line, "3",
                "--trailing-comma",
                "--use-parentheses",
            ]
-   repo: https://github.com/PyCQA/flake8
    rev: 6.0.0
    hooks:
    -   id: flake8
        language_version: python3
        additional_dependencies:
            - flake8-bugbear
            - flake8-comprehensions
            - flake8-debugger
            - flake8-string-format
        exclude: django_app/frontend/views.py
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
    -   id: check-executables-have-shebangs
    -   id: check-yaml
    -   id: end-of-file-fixer
    -   id: trailing-whitespace
    -   id: detect-private-key
    -   id: name-tests-test
        args: [--django]
```

Comment (justin-sleep) [reply .pre-commit-config.yaml:27]: Those are good suggestions. After thinking about it a bit further, and to help consolidate the config, I've switched to [Ruff](https://github.com/astral-sh/ruff), which provides most of these features out of the box (and is quite a bit faster).

File: .pre-commit-config.yaml
Comment (rgasper) [inline .pre-commit-config.yaml:1]: could we keep `black` in here tho? I don't think ruff replaces that

Comment (rgasper) [reply .pre-commit-config.yaml:1]: https://github.com/astral-sh/ruff-pre-commit states that `black` and `isort` should come after `ruff`

Comment (justin-sleep) [reply .pre-commit-config.yaml:1]: Yeah, looks like Ruff is working on autoformatting but [isn't there yet](https://github.com/astral-sh/ruff/issues/1904), so I'll re-add Black.

**Comment by rgasper** [comment]
Welp we didn't move fast enough and now there's a lot of drift

Let's just add the `.pre-commit-config.yaml` directly to main after we merge the rest of the recent new feature / bugfix PRs?

**Comment by justin-sleep** [comment]
Yeah, let's merge those PRs and then I'll clear up the merge conflicts. There are unfortunately a bunch of manual changes that the linter couldn't autofix, so it'll be faster to handle the merge conflicts than to start from scratch

**Pull request merged**